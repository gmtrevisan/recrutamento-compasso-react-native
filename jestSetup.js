import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { TestScheduler } from 'rxjs/testing';
import { render } from '@testing-library/react-native';
import { reducers } from './src/redux/store';
import { initialState as initialStateNews } from './src/redux/news/reducer';
import { initialState as initialStateModal } from './src/redux/modal/reducer';

const config = (initialState) => {
    const store = createStore(reducers, initialState);
    store.dispatch = jest.fn();
    return store;
};

export const stateEpic = {
    value: config({
        newsReducer: { ...initialStateNews },
        modalReducer: { ...initialStateModal },
    }).getState(),
};

export const createTestScheduler = () => new TestScheduler((actual, expected) => {
    expect(actual).toEqual(expected);
});

export function renderWithRedux(ui, { initialState = {}, store = config(initialState) } = {}) {
    return {
        ...render(<Provider store={store}>{ui}</Provider>),
        store,
    };
};
