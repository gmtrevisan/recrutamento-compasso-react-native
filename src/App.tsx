import React from 'react';
import { StatusBar } from 'expo-status-bar';
import styled from 'styled-components/native';
import { Provider } from 'react-redux';
import { Header } from './components/Header/Header';
import { NewsList } from './components/NewsList/NewsList';
import { store } from './redux/store';
import { NewsDetails } from './components/NewsDetails/NewsDetails';

export const App = () => {
  return (
    <Provider store={store}>
      <Container>
        <Header />
        <NewsDetails />
        <NewsList />
        <StatusBar style="auto" />
      </Container>
    </Provider>
  )
}

const Container = styled.View`
  flex: 1;
  background-color: #fff;
  padding-top: 22px;
`;

export default App;
