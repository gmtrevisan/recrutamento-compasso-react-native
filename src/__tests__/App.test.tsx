import React from 'react';
import { renderWithRedux } from '../../jestSetup';
import { App } from '../App';

describe('App.test.tsx', () => {
  const news01 = {
    _id: '123',
    headline: {
      main: 'NewsTest1'
    },
    web_url: 'WebUrl',
    abstract: 'Abstract',
  };

  const mockNewsState = {
    loading: false,
    error: false,
    news: [news01],
    page: 0,
  };

  const mockModalState = {
    open: false,
    news: news01,
  };

  test('render buttons', () => {
    const { toJSON } = renderWithRedux(<App />, { initialState: { modalReducer: mockModalState, newsReducer: mockNewsState } });
    expect(toJSON()).toMatchSnapshot();
  });
});