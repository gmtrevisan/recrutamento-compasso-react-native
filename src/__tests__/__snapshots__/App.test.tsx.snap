// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.test.tsx render buttons 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingTop": 22,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "display": "flex",
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#480087",
          "borderBottomWidth": 5,
          "borderColor": "#FFB1A3",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "opacity": 1,
          "paddingBottom": 20,
          "paddingTop": 20,
          "textAlign": "center",
        }
      }
    >
      <Text
        selected={true}
        style={
          Array [
            Object {
              "color": "#FFB1A3",
              "textAlign": "center",
            },
          ]
        }
      >
        CiÃªncia
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#480087",
          "borderBottomWidth": 5,
          "borderColor": "#663a8c",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "opacity": 1,
          "paddingBottom": 20,
          "paddingTop": 20,
          "textAlign": "center",
        }
      }
    >
      <Text
        selected={false}
        style={
          Array [
            Object {
              "color": "#FFF",
              "textAlign": "center",
            },
          ]
        }
      >
        Tecnologia
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "paddingLeft": 20,
          "paddingRight": 20,
          "textAlign": "center",
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#FFF",
          "opacity": 1,
          "paddingBottom": 10,
          "paddingTop": 10,
          "textAlign": "center",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "lineHeight": 20,
              "textAlign": "center",
            },
          ]
        }
      >
        Erro ao carregar, espere alguns segundos e tente novamente.
      </Text>
    </View>
  </View>
</View>
`;
